python/samples/getting_started_with_agents/multi_agent_orchestration/step3b_group_chat_with_chat_completion_manager.py

```diff
@@ -0,0 +1,368 @@
+    artist = ChatCompletionAgent(
+        name="Artist",
+        description="An artist from Africa.",
+        instructions=(
+            "You're an artist from Africa. "
+            "You view life through creative expression, storytelling, and collective memory. "
+            "You are in a debate. Feel free to challenge the other participants with respect."
+        ),
+        service=AzureChatCompletion(),
+    )
+    immigrant = ChatCompletionAgent(
+        name="Immigrant",
+        description="An immigrant entrepreneur from Asia living in Canada.",
+        instructions=(
+            "You're an immigrant entrepreneur from Asia living in Canada. "
+            "You balance trandition with adaption. "
+            "You focus on family success, risk, and opportunity. "
+            "You are in a debate. Feel free to challenge the other participants with respect."
+        ),
+        service=AzureChatCompletion(),
+    )
+    doctor = ChatCompletionAgent(
+        name="Doctor",
+        description="A doctor from Scandinavia.",
+        instructions=(
+            "You're a doctor from Scandinavia. "
+            "Your perspective is shaped by public health, equity, and structured societal support. "
+            "You are in a debate. Feel free to challenge the other participants with respect."
+        ),
+        service=AzureChatCompletion(),
+    )
+
+    return [farmer, developer, teacher, activist, spiritual_leader, artist, immigrant, doctor]
+
+
+class ChatCompletionGroupChatManager(GroupChatManager):
+    """A simple chat completion base group chat manager.
+
+    This chat completion service requires a model that supports structured output.
+    """
+
+    service: ChatCompletionClientBase
+
+    topic: str
+
+    termination_prompt: str = (
+        "You are mediator that guides a discussion on the topic of '{{$topic}}'. "
+        "You need to determine if the discussion has reached a conclusion. "
+        "If you would like to end the discussion, please respond with True. Otherwise, respond with False."
+    )
+
+    selection_prompt: str = (
+        "You are mediator that guides a discussion on the topic of '{{$topic}}'. "
+        "You need to select the next participant to speak. "
+        "Here are the names and descriptions of the participants: "
+        "{{$participants}}\n"
+        "Please respond with only the name of the participant you would like to select."
+    )
+
+    result_filter_prompt: str = (
+        "You are mediator that guides a discussion on the topic of '{{$topic}}'. "
+        "You have just concluded the discussion. "
+        "Please summarize the discussion and provide a closing statement."
+    )
+
+    def __init__(self, topic: str, service: ChatCompletionClientBase, **kwargs) -> None:
+        """Initialize the group chat manager."""
+        super().__init__(topic=topic, service=service, **kwargs)
+
+    async def _render_prompt(self, prompt: str, arguments: KernelArguments) -> str:
```

Comment:
I think it would greatly help devs if we provide more comments around the user of these methods. For example, when/why do I need to override `_render_prompt`?